### PR TITLE
feat(vg_lite): add stroke path support

### PR DESCRIPTION
**Pull request by FASTSHIFT**
### Description of the feature or fix

Add stroke path support
image

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by XuNeo — DISMISSED** [review]

File: src/draw/vg_lite/lv_draw_vg_lite_vector.c [hunk 1]
```diff
@@ -336,6 +343,43 @@ static void lv_path_to_vg(lv_vg_lite_path_t * dest, const lv_vector_path_t * src
...
+            lv_stroke_join_to_vg(dsc->join),
+            dsc->width,
+            dsc->miter_limit,
+            lv_array_front(&dsc->dash_pattern),
+            dsc->dash_pattern.size,
+            dsc->width / 2,
+            lv_color32_to_vg(dsc->color, dsc->opa))
+    );
+
+    if(dsc->style == LV_VECTOR_DRAW_STYLE_SOLID) {
```
Comment (PGNetHun) [inline src/draw/vg_lite/lv_draw_vg_lite_vector.c]: I guess this if-else is here to be extended later. 
Could you please add `// TODO: implement .... ` or remove if-else, if not needed?
In my opinion it is not a problem to place TODOs in source code, later devs can go through them to prioritize which is more important to implement next, or can be omitted totally.

File: src/others/vg_lite_tvg/vg_lite_tvg.cpp [hunk 2]
```diff
@@ -711,6 +713,67 @@ extern "C" {
...
+        if(!path || line_width <= 0) {
+            return VG_LITE_INVALID_ARGUMENT;
+        }
+
+        if(miter_limit < 1.0f) {
+            miter_limit = 1.0f;
+        }
+
+        if(!path->stroke) {
+            path->stroke = (vg_lite_stroke_t *)malloc(sizeof(vg_lite_stroke_t));
```
Comment (PGNetHun) [inline src/others/vg_lite_tvg/vg_lite_tvg.cpp]: Please use `lv_malloc`

File: src/others/vg_lite_tvg/vg_lite_tvg.cpp [hunk 3]
```diff
@@ -820,7 +883,13 @@ extern "C" {
 
     vg_lite_error_t vg_lite_clear_path(vg_lite_path_t * path)
     {
-        LV_UNUSED(path);
+        LV_ASSERT_NULL(path);
+
+        if(path->stroke) {
+            free(path->stroke);
```
Comment (PGNetHun) [inline src/others/vg_lite_tvg/vg_lite_tvg.cpp]: Please use `lv_free`

File: src/others/vg_lite_tvg/vg_lite_tvg.cpp [hunk 4]
```diff
@@ -711,6 +713,67 @@ extern "C" {
...
+        }
+
+        if(!path->stroke) {
+            path->stroke = (vg_lite_stroke_t *)malloc(sizeof(vg_lite_stroke_t));
+
+            if(!path->stroke) {
+                return VG_LITE_OUT_OF_RESOURCES;
+            }
+
+            memset(path->stroke, 0, sizeof(vg_lite_stroke_t));
```
Comment (PGNetHun) [inline src/others/vg_lite_tvg/vg_lite_tvg.cpp]: Please use `lv_memset`
(maybe in a specific platform the implementation is faster / more optimized than the default `memset` )

**Review by PGNetHun — CHANGES_REQUESTED** [review]

File: src/draw/vg_lite/lv_draw_vg_lite_vector.c (hunk 1 above)
Comment (FASTSHIFT) [reply src/draw/vg_lite/lv_draw_vg_lite_vector.c]: This judgment has no effect, I will delete it.

**Review by PGNetHun — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]

**Pull request merged**